pkg/executor/bacalhau/client.go

```diff
@@ -0,0 +1,70 @@
+package bacalhau
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/bacalhau-project/bacalhau/pkg/publicapi/apimodels"
+	retryablehttp "github.com/hashicorp/go-retryablehttp"
+)
+
+const API_VERSION = "v1"
+
+type BacalhauClient struct {
+	http    *retryablehttp.Client
+	options BacalhauExecutorOptions
+}
+
+func NewClient(options BacalhauExecutorOptions) (*BacalhauClient, error) {
```

Comment:
```suggestion
func newClient(options BacalhauExecutorOptions) (*BacalhauClient, error) {
```
Could we make this private? We aren't using it outside of the `bacalhau` package.

Reply:
Good call! made all client functions private